### PR TITLE
prometheus: fix the chart to correctly reference local dependencies

### PR DESCRIPTION
**Pull request by dkoshkin**
Part of https://jira.mesosphere.com/browse/DCOS-61056

Although the change looks large, the prometheus chart was never using those versions.

To minimize the impact I untarred the versions that were already there, fixed the api versions and fixed the `requirements.yaml` file to reference those charts

Tested locally:
```
➜  dkkonvoyv11633 k get nodes
NAME                                         STATUS   ROLES    AGE    VERSION
ip-10-0-128-25.us-west-2.compute.internal    Ready    <none>   169m   v1.16.3
ip-10-0-130-221.us-west-2.compute.internal   Ready    <none>   169m   v1.16.3
ip-10-0-130-59.us-west-2.compute.internal    Ready    <none>   169m   v1.16.3
ip-10-0-131-181.us-west-2.compute.internal   Ready    <none>   169m   v1.16.3
ip-10-0-193-100.us-west-2.compute.internal   Ready    master   170m   v1.16.3
```
```
➜  dkkonvoyv11633 helm list
prometheus-kubeaddons                    	1       	Wed Nov 20 08:24:59 2019	DEPLOYED	prometheus-operator-5.19.6  	0.31.1      	kubeaddons
```
```
kubeaddons              prometheus-kubeaddons-grafana-5b57498799-585pp                       2/2     Running     0          48m
kubeaddons              prometheus-kubeaddons-kube-state-metrics-7d75c49cc9-65zpc            1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prom-operator-74cff8df9b-bttbd                 1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prometheus-node-exporter-5gvsr                 1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prometheus-node-exporter-6sw2w                 1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prometheus-node-exporter-n57jq                 1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prometheus-node-exporter-vx9vt                 1/1     Running     0          48m
kubeaddons              prometheus-kubeaddons-prometheus-node-exporter-w9kv2                 1/1     Running     0          48m
kubeaddons              prometheus-prometheus-kubeaddons-prom-prometheus-0                   5/5     Running     2          47m
kubeaddons              prometheusadapter-kubeaddons-prometheus-adapter-7b8cc77cf5zfgxw      1/1     Running     0          47m
```

### CODE REVIEW

**Comment by gracedo** [comment]
@dkoshkin just wondering what was the reasoning behind using local chart deps? it's been on my mind to upgrade our fork to the latest official release and I'm just thinking ahead to how difficult that upgrade/rebase might get with all of these local chart deps which likewise will need to be updated with every upgrade 🤔

**Comment by dkoshkin** [comment]
@gracedo the immediate motivation is that we need to upgrade to Kubernetes v1.16.
I looked into upgrading the `prometheus` chart to a version that has all the dependencies recent enough that also fix the api versioning but that ended up with alot of changes, I didn't have the bandwidth or experience to apply the changes that we have added and to test it all properly.

This change was the minimal change to get the k8s v1.16 work unblocked.

I agree though, we have lots of chart forks here, its getting pretty difficult to maintain.

File: staging/prometheus-operator/dependent-charts/kube-state-metrics/values.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Default values for kube-state-metrics.
 prometheusScrape: true
 image:
-  repository: k8s.gcr.io/kube-state-metrics
-  tag: v1.5.0
+  repository: quay.io/coreos/kube-state-metrics
```
Comment (shaneutt) [inline staging/prometheus-operator/dependent-charts/kube-state-metrics/values.yaml]: I take it this change reflects upstream changes?

**Comment by gracedo** [comment]
@dkoshkin thanks for explaining! What is the version you found that has all the deps recent enough for upgrading to v1.16? Just want to make a note in the upgrade [JIRA ticket](https://jira.mesosphere.com/browse/DCOS-60484) to be sure to upgrade to a version above that one and hopefully we can remove `/dependent-charts` at that time as well :)

File: staging/prometheus-operator/dependent-charts/kube-state-metrics/values.yaml (hunk 1 above)
Comment (dkoshkin) [reply staging/prometheus-operator/dependent-charts/kube-state-metrics/values.yaml]: Yes all the changes that are in the PR is what was already being deployed as part of `master`.

**Comment by dkoshkin** [comment]
@gracedo this PR didn't change the versions of the charts in `/dependent-charts` its just now correctly referencing those versions.
I made minor changes to api versions, which was the intention of https://github.com/mesosphere/charts/commit/00ecffb8379b316cec451abfcedd56f71417fb69 and https://github.com/mesosphere/charts/commit/5d7d580ad348bbe2ea654bd8fdb8ccb5f4451c6f.

But because the changes were made to the dependent charts without actually referencing those charts, those were never reflected.

**Comment by gracedo** [comment]
@dkoshkin ah! ok I understand now 😄do you know the original intention behind using local chart deps? I think I saw they were added at the start when we first forked the chart, but we didn't seem to need them as they were never referenced (until now) and things were working as expected. just trying to get the full picture so when we eventually upgrade prometheus-operator to the latest official version (assuming its deps have all the updated api versions), we can decide if it makes sense to get rid of local chart deps

**Pull request merged**